apps/flame-defi/app/pool/context/pool-position-context.tsx

```diff
@@ -1,132 +1,264 @@
+  createPoolService,
+} from "features/evm-wallet";
+import { useAccount, useConfig } from "wagmi";
+import { useEvmChainData } from "config/hooks/use-config";
+import {
+  getTokensLiquidityAmounts,
+  getMinMaxTick,
+  getTokenDataFromCurrencies,
+  sqrtPriceX96ToPrice,
+  tickToPrice,
+} from "./pool-position-helpers";
 
 export const PoolPositionContext = createContext<
   PoolPositionContextProps | undefined
 >(undefined);
 
-const mockPosition: Positions = {
-  0: {
-    tokens: [
-      {
-        symbol: "TIA",
-        unclaimedFees: 0.0078,
-        liquidity: 1.13314,
-        liquidityPercentage: 0.05,
-      },
-      {
-        symbol: "milkTIA",
-        unclaimedFees: 0.00123,
-        liquidity: 0.8814,
-        liquidityPercentage: 0.05,
-      },
-    ],
-    feeTier: 0.0005,
-    inRange: true,
-    min: 0,
-    max: "∞",
-    positionStatus: "In range",
-  },
-  1: {
-    tokens: [
-      {
-        symbol: "TIA",
-        unclaimedFees: 0.0078,
-        liquidity: 1.13314,
-        liquidityPercentage: 0.05,
-      },
-      {
-        symbol: "USDC",
-        unclaimedFees: 0.00123,
-        liquidity: 0.8814,
-        liquidityPercentage: 0.05,
-      },
-    ],
-    feeTier: 0.0005,
-    inRange: false,
-    min: 0,
-    max: "∞",
-    positionStatus: "Closed",
-  },
-};
-
-const defaultPoolToken = {
-  symbol: "",
-  unclaimedFees: 0,
-  liquidity: 0,
-  liquidityPercentage: 0,
-};
-
 export const PoolPositionContextProvider = ({
   children,
 }: PropsWithChildren) => {
   const params = useParams();
+  const wagmiConfig = useConfig();
+  const { address } = useAccount();
   const { formatNumber } = useIntl();
+  const tokenId = params["token-id"] as string;
+  const { selectedChain } = useEvmChainData();
+  const { currencies } = selectedChain;
   const currentPoolSettings = getFromLocalStorage("poolSettings") || {};
   const [collectAsNative, setCollectAsNative] = useState<boolean>(
     currentPoolSettings.collectAsNative || false,
   );
-  const poolId = params["pool-id"];
-  const position = mockPosition[Number(poolId)];
-  const [positionDetails, setPositionDetails] = useState<Position | undefined>(
-    position,
-  );
-  const symbols = positionDetails?.tokens.map((token) => token.symbol) || [];
-  const [poolTokens, setPoolTokens] = useState<PoolToken[]>(
-    positionDetails?.tokens || [],
-  );
+  const [currentPrice, setCurrentPrice] = useState<string>("");
+  const [minPrice, setMinPrice] = useState<string>("");
+  const [maxPrice, setMaxPrice] = useState<string>("");
+  const [invertedPrice, setInvertedPrice] = useState<boolean>(false);
+  const [symbols, setSymbols] = useState<string[]>([]);
+  const [poolTokens, setPoolTokens] = useState<PoolToken[] | []>([]);
+  const [feeTier, setFeeTier] = useState<string>("");
 
-  const poolTokenOne = poolTokens[0] || defaultPoolToken;
-  const poolTokenTwo = poolTokens[1] || defaultPoolToken;
+  const poolTokenOne = poolTokens[0] || null;
```

Comment:
Why is there `poolTokens` and `poolTokenOne`/`Two`?

Reply:
poolTokens is used pretty heavily in the poolTxnSteps as well as some hooks and other places where it's more advantageous to loop over the values. It's be preferable to get rid of the poolTokenOne and poolTokenTwo at the context level than to get rid of the poolTokens array. But I'd probably need to de-structure poolTokens in a couple places

Reply:
If they're always the same, ie poolTokens = [poolTokenOne, poolTokenTwo], I don't think it makes sense to pass multiple data structures around.

Where it's advantageous to loop just do `[poolTokenOne, poolTokenTwo].map(...)`